ruby/ql/lib/codeql/ruby/ast/internal/Synthesis.qll

```diff
@@ -950,6 +950,40 @@ private module DestructuredAssignDesugar {
     }
   }
 
+  abstract private class LhsWithReceiver extends Expr {
+    abstract Expr getReceiver();
+
+    abstract SynthKind getSynthKind();
+  }
+
+  private class LhsCall extends LhsWithReceiver instanceof MethodCall {
+    final override Expr getReceiver() { result = MethodCall.super.getReceiver() }
+
+    final override SynthKind getSynthKind() {
+      result = MethodCallKind(super.getMethodName(), false, super.getNumberOfArguments())
+    }
+  }
+
+  private class LhsScopedConstant extends LhsWithReceiver, TScopeResolutionConstantAccess {
```

Comment:
I think it would be better to move the existing (private) `ScopeResolutionConstantAccess` class into `internal/Constant.qll`, and then reference that here.